pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/groupby/AggregationGroupByTrimmingService.java

```diff
@@ -115,9 +115,15 @@ public AggregationGroupByTrimmingService(@Nonnull AggregationFunction[] aggregat
           if (priorityQueue == null) {
             trimmedResults.get(i).put(groupKey, intermediateResults[i]);
           } else {
-            priorityQueue.add(new GroupKeyResultPair(groupKey, (Comparable) intermediateResults[i]));
-            if (priorityQueue.size() > _trimSize) {
-              priorityQueue.poll();
+            GroupKeyResultPair newValue = new GroupKeyResultPair(groupKey, (Comparable) intermediateResults[i]);
+            if (priorityQueue.size() >= _trimSize) {
```

Comment:
_trimSize is set to a min of 5000 in the current code. Nonetheless, I have added a null check against topValue (will happen when the queue is empty).